boot/duneboot.ml

```diff
@@ -1223,27 +1253,7 @@ module File_kind = struct
         else None, None, `C_comp
       in
       Some (Asm { syntax; arch; os; assembler })
-    | ".c" ->
-      let arch, flags =
-        let fn = Filename.remove_extension fn in
-        let check suffix = String.ends_with fn ~suffix in
-        let x86 gnu _msvc =
-          (* CR rgrinberg: select msvc flags on windows *)
-          Some `X86, gnu
-        in
-        if check "_sse2"
-        then x86 [ "-msse2" ] [ "/arch:SSE2" ]
-        else if check "_sse41"
-        then x86 [ "-msse4.1" ] [ "/arch:AVX" ]
-        else if check "_avx2"
-        then x86 [ "-mavx2" ] [ "/arch:AVX2" ]
-        else if check "_avx512"
-        then x86 [ "-mavx512f"; "-mavx512vl"; "-mavx512bw" ] [ "/arch:AVX512" ]
-        else if String.ends_with fn ~suffix:"_neon"
-        then Some `Arm64, []
-        else None, []
-      in
-      Some (C { arch; flags })
+    | ".c" -> Some (C { cpu_feature = Cpu_feature.by_suffix fname })
```

Comment:
Why do we need to discover the "cpu feature" here? Why don't we do it with the rest of the flags?

Reply:
This is the "cpu feature" that the source file is asking for. Whether or not we can provide it is decided later.

Reply:
That's exactly the problem. If we can only decide what to do with this cpu feature later, why are we discovering it now? Might as well discover it when we need it.